zaza/openstack/charm_tests/ceilometer_agent/tests.py

```diff
@@ -0,0 +1,116 @@
+        services = []
+
+        if current_release >= CeilometerAgentTest.XENIAL_PIKE:
+            services.append('ceilometer-polling: AgentManager worker(0)')
+            services.append(
+                'ceilometer-agent-notification: NotificationService worker(0)')
+        elif current_release >= CeilometerAgentTest.XENIAL_OCATA:
+            services.append('ceilometer-collector: CollectorService worker(0)')
+            services.append('ceilometer-polling: AgentManager worker(0)')
+            services.append(
+                'ceilometer-agent-notification: NotificationService worker(0)')
+            services.append('apache2')
+        elif current_release >= CeilometerAgentTest.XENIAL_NEWTON:
+            services.append('ceilometer-collector - CollectorService(0)')
+            services.append('ceilometer-polling - AgentManager(0)')
+            services.append(
+                'ceilometer-agent-notification - NotificationService(0)')
+            services.append('ceilometer-api')
+        else:
+            services.append('ceilometer-collector')
+            services.append('ceilometer-api')
+            services.append('ceilometer-agent-notification')
+
+            if current_release < CeilometerAgentTest.TRUSTY_MITAKA:
+                services.append('ceilometer-alarm-notifier')
+                services.append('ceilometer-alarm-evaluator')
+
+            if current_release >= CeilometerAgentTest.TRUSTY_LIBERTY:
+                # Liberty and later
+                services.append('ceilometer-polling')
+            else:
+                # Juno and earlier
+                services.append('ceilometer-agent-central')
+
+        return services
+
+    # NOTE(beisner): need to add more functional tests
+
+    def test_900_restart_on_config_change(self):
+        """Checking restart happens on config change."""
+        # Expected default and alternate values
+        current_value = openstack_utils.get_application_config_option(
+            'ceilometer', 'debug'
+        )
+        assert type(current_value) == bool
+        new_value = not current_value
+
+        # Convert bool to str
+        current_value = str(current_value)
+        new_value = str(new_value)
+
+        set_default = {'debug': current_value}
+        set_alternate = {'debug': new_value}
+        default_entry = {'DEFAULT': {'debug': [current_value]}}
+        alternate_entry = {'DEFAULT': {'debug': [new_value]}}
+
+        logging.info('changing config: {}'.format(set_alternate))
+        self.restart_on_changed(
+            CeilometerAgentTest.CONF_FILE,
+            set_default,
+            set_alternate,
+            default_entry,
+            alternate_entry,
+            self.services)
+
+    def test_901_pause_resume(self):
+        """Run pause and resume tests.
+
+        Pause service and check services are stopped then resume and check
+        they are started.
+        """
+        self.pause_resume(self.services)
```

Comment:
This will not run the tests. pause_resume is a contextmanager, calling it needs be called as such "with self.pause_resume(...)